components/console_cmd_ifconfig/console_ifconfig.c

```diff
@@ -0,0 +1,637 @@
+{
+    esp_netif_ip_info_t ip_info;
+
+    esp_netif_dhcpc_stop(esp_netif);
+    memset(&ip_info, 0, sizeof(esp_netif_ip_info_t));
+    esp_netif_get_ip_info(esp_netif, &ip_info);
+
+    if (!strcmp("ip", argv[self->start_index])) {
+        ESP_LOGI(TAG, "Setting ip: %s", argv[self->start_index + 1]);
+
+        inet_aton(argv[self->start_index + 1], &ip_info.ip.addr);
+        esp_netif_set_ip_info(esp_netif, &ip_info);
+        return ESP_OK;
+    } else if (!strcmp("mask", argv[self->start_index])) {
+        ESP_LOGI(TAG, "Setting mask: %s", argv[self->start_index + 1]);
+
+        inet_aton(argv[self->start_index + 1], &ip_info.netmask.addr);
+        esp_netif_set_ip_info(esp_netif, &ip_info);
+        return ESP_OK;
+    } else if (!strcmp("gw", argv[self->start_index])) {
+        ESP_LOGI(TAG, "Setting gw: %s", argv[self->start_index + 1]);
+
+        inet_aton(argv[self->start_index + 1], &ip_info.gw.addr);
+        esp_netif_set_ip_info(esp_netif, &ip_info);
+        return ESP_OK;
+    }
+
+    return ESP_FAIL;
+}
+
+
+#if IP_NAPT
+esp_err_t set_napt(char *if_name, bool state)
+{
+    esp_netif_t *esp_netif = NULL;
+    esp_err_t ret = ESP_FAIL;
+    char interface[10];
+
+    /* Get interface details and own global ipv6 address */
+    for (int i = 0; i < esp_netif_get_nr_of_ifs(); ++i) {
+        esp_netif = esp_netif_next(esp_netif);
+
+        ret = esp_netif_get_netif_impl_name(esp_netif, interface);
+        if ((ESP_FAIL == ret) || (NULL == esp_netif)) {
+            ESP_LOGE(TAG, "No interface available");
+            return ESP_FAIL;
+        }
+
+        if (!strcmp(interface, if_name)) {
+            struct netif *lwip_netif = esp_netif_get_netif_impl(esp_netif);
+            ip_napt_enable_netif(lwip_netif, state);
+            return ESP_OK;
+        }
+    }
+
+    return ESP_FAIL;
+}
+#endif
+
+
+esp_err_t ifcfg_napt_op(netif_op *self, int argc, char *argv[], esp_netif_t *esp_netif)
+{
+#if IP_NAPT
+    if (!strcmp("napt", argv[self->start_index])) {
+
+        ESP_LOGI(TAG, "Setting napt %s on %s", argv[self->start_index + 1], argv[1]);
+        if (!strcmp(argv[self->start_index + 1], "enable")) {
+            return set_napt(argv[1], true);
+        } else if (!strcmp(argv[self->start_index + 1], "disable")) {
+            return set_napt(argv[1], false);
+        } else {
+            ESP_LOGI(TAG, "Invalid argument: %s", argv[self->start_index + 1]);
+        }
+
+        return ESP_FAIL;
+    }
+#endif
+    ESP_LOGE(TAG, "NAPT not enabled in menuconfig");
+    return ESP_OK;
+}
+
+
+esp_err_t ifcfg_dhcp_op(netif_op *self, int argc, char *argv[], esp_netif_t *esp_netif)
+{
+    if (!strcmp("server", argv[self->start_index + 1])) { // Server
+        if (!strcmp("enable", argv[self->start_index + 2])) {
+            ESP_LOGW(TAG, "DHCP Server configuration is not supported yet.");    // TBD
+            //esp_netif_dhcps_start(esp_netif);
```

Comment:
Should we uncomment?

Reply:
The dhcp server doesn't work with the runtime console component.
This has to be fixed in IDF. So for now I'll keep this line commented.